app/main.py

```diff
@@ -1,25 +1,67 @@
 from abc import ABC
+from typing import Any
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Any, name: str) -> None:
+        self.name = f"_{name}"
+
+    def __get__(self, instance: Any, owner: Any) -> Any:
+        return getattr(instance, self.name)
+
+    def __set__(self, instance: Any, value: int) -> None:
+        if (not isinstance(self.min_amount, int)
+                and isinstance(self.max_amount, int)):
```

Comment:
The condition `not isinstance(self.min_amount, int) and isinstance(self.max_amount, int)` seems incorrect. It should check both `min_amount` and `max_amount` to be integers using `or` instead of `and`. The correct condition should be `not (isinstance(self.min_amount, int) and isinstance(self.max_amount, int))`.